From b00d01a86b652af43ccb2d40756296ece2384c93 Mon Sep 17 00:00:00 2001
From: Davor Joja <davorjoja@logicbricks.com>
Date: Wed, 5 Nov 2014 16:50:59 +0100
Subject: [PATCH 0445/1566] gpu: drm: xylon: Fixed compile error

Replace call to drm_fb_helper_restore_fbdev_mode() with
drm_fb_helper_restore_fbdev_mode_unlocked().

Error log:
drivers/gpu/drm/xylon/xylon_fbdev.c: In function 'xylon_drm_fbdev_restore_mode':
drivers/gpu/drm/xylon/xylon_fbdev.c:218:3: error: implicit declaration of
function 'drm_fb_helper_restore_fbdev_mode'
[-Werror=implicit-function-declaration]
   drm_fb_helper_restore_fbdev_mode(&fbdev->fb_helper);
   ^

Signed-off-by: Davor Joja <davorjoja@logicbricks.com>
Tested-by: Christian Kohn <christian.kohn@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 drivers/gpu/drm/xylon/xylon_fbdev.c |    4 +---
 1 files changed, 1 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/xylon/xylon_fbdev.c b/drivers/gpu/drm/xylon/xylon_fbdev.c
index 196e0b0..0d848c7 100644
--- a/drivers/gpu/drm/xylon/xylon_fbdev.c
+++ b/drivers/gpu/drm/xylon/xylon_fbdev.c
@@ -212,9 +212,7 @@ void xylon_drm_fbdev_restore_mode(struct xylon_drm_fb_device *fbdev)
 	if (fbdev) {
 		dev = fbdev->fb_helper.dev;
 
-		drm_modeset_lock_all(dev);
-		drm_fb_helper_restore_fbdev_mode(&fbdev->fb_helper);
-		drm_modeset_unlock_all(dev);
+		drm_fb_helper_restore_fbdev_mode_unlocked(&fbdev->fb_helper);
 	}
 }
 
-- 
1.7.5.4

